app/src/main/java/com/graddex/graddex/PokemonDetailsFragment.kt

```diff
@@ -0,0 +1,96 @@
+
+        Log.d(tag, "PokemonID: $pokemonId")
+
+        // Inflate the Pokemon Details fragment layout
+        binding = DataBindingUtil.inflate<FragmentPokemonDetailsBinding>(
+            inflater, R.layout.fragment_pokemon_details, container, false
+        )
+
+        // Call the Pokemon Details view model for specified Pokemon ID
+        viewModel = ViewModelProvider(this)
+            .get(PokemonDetailsViewModel::class.java)
+
+        binding.statusText.text = getString(loading)
+        viewModel.syncPokemonDetails(pokemonId)
+
+        return binding.root
+
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        // Update UI with details
+        viewModel.pokemonSpriteFront.observe(viewLifecycleOwner,
+            { pokemonSpriteFront ->
+                binding.pokemonFrontImage.load(pokemonSpriteFront)
+                binding.statusText.visibility = View.GONE
+            })
+        viewModel.pokemonSpriteBack.observe(viewLifecycleOwner,
+            { pokemonSpriteBack -> binding.pokemonBackImage.load(pokemonSpriteBack) })
+        viewModel.pokemonName.observe(viewLifecycleOwner,
+            { pokemonName -> binding.pokemonName.text = pokemonName.toString().capitalize(Locale.ROOT) })
+        viewModel.pokemonTypes.observe(viewLifecycleOwner,
+            { pokemonTypes ->
+                val pokemonTypeList = mutableListOf<String>()
+                for (element in pokemonTypes) {
+                    pokemonTypeList += element.type.name.capitalize(Locale.ROOT)
+                }
+                binding.pokemonTypes.text = pokemonTypeList.joinToString(" | ")
+            })
+        viewModel.pokemonAbilities.observe(viewLifecycleOwner,
+            { pokemonAbilities ->
+                Log.d(tag, "Pokemon Abilities: $pokemonAbilities")
+                val pokemonAbilitiesList = mutableListOf<String>()
+                for (element in pokemonAbilities) {
+                    if (!element.is_hidden) {
+                        pokemonAbilitiesList += element.ability.name.capitalize(Locale.ROOT)
+                    } else {
+                        val pokemonHiddenAbility = element.ability.name.capitalize(Locale.ROOT)
+                        val hiddenAbilityText = String.format("%s %s", getString(hidden_ability), pokemonHiddenAbility)
+                        binding.pokemonHiddenAbility.text = hiddenAbilityText
+                    }
+                }
+                if (pokemonAbilitiesList.size > 1) {
+                    val pokemonAbilityText = String.format("%s %s", getString(abilities), pokemonAbilitiesList.joinToString(" and "))
+                    binding.pokemonAbilities.text = pokemonAbilityText
+                } else {
+                    val pokemonAbilityText = String.format("%s %s", getString(ability), pokemonAbilitiesList[0])
+                    binding.pokemonAbilities.text = pokemonAbilityText
+                }
```

Comment:
This is a bit more of an FYI, you can actually just do this:

```suggestion
                val pokemonAbilityText = String.format("%s %s", getString(abilities), pokemonAbilitiesList.joinToString(" and "))
                binding.pokemonAbilities.text = pokemonAbilityText
```
This works because `joinToString` is a bit smart and It'll only put the `" and "` if there's more than 1 thing in the list :)

Reply:
I had it this way to just change whether it would say "Ability: " or "Abilities: " depending on whether there was more than one ability :) (just my OCD haha)

Reply:
Ah, I gotcha! 

This will be good to look into in later, check out something called "plurals" or "quantity strings": https://medium.com/@info.anikdey003/android-quantity-strings-plurals-5ed3d7c62c16